cibuildwheel/platforms/android.py

```diff
@@ -0,0 +1,429 @@
+        os.environ.update(self.android_env)
+
+        pth_file = self.site_packages / "_cross_venv.pth"
+        pth_file.write_text("import _cross_venv; _cross_venv.initialize()")
+
+        try:
+            yield
+        finally:
+            pth_file.unlink()
+            for key, original_value in original_env.items():
+                if original_value is None:
+                    del os.environ[key]
+                else:
+                    os.environ[key] = original_value
+
+    def before_build(self) -> None:
+        if self.build_options.before_build:
+            log.step("Running before_build...")
+            shell_prepared(self.build_options.before_build, self.build_options, self.env)
+
+    def build_wheel(self) -> Path:
+        log.step("Building wheel...")
+        built_wheel_dir = self.tmp_dir / "built_wheel"
+        with self.simulate_android():
+            call(
+                "python",
+                "-m",
+                "build",
+                self.build_options.package_dir,
+                "--wheel",
+                "--no-isolation",
+                "--skip-dependency-check",
+                f"--outdir={built_wheel_dir}",
+                *get_build_frontend_extra_flags(
+                    self.build_frontend,
+                    self.build_options.build_verbosity,
+                    self.build_options.config_settings,
+                ),
+            )
+
+        built_wheels = list(built_wheel_dir.glob("*.whl"))
+        if len(built_wheels) != 1:
+            msg = f"{built_wheel_dir} contains {len(built_wheels)} wheels; expected 1"
+            raise errors.FatalError(msg)
+        built_wheel = built_wheels[0]
+
+        if built_wheel.name.endswith("none-any.whl"):
+            raise errors.NonPlatformWheelError()
+        return built_wheel
+
+    def test_wheel(self, wheel: Path) -> None:
+        if not (
+            self.build_options.test_command
+            and self.build_options.test_selector(self.config.identifier)
+        ):
+            return
+
+        log.step("Testing wheel...")
+        native_arch = arch_synonym(platform.machine(), native_platform(), "android")
+        if self.config.arch != native_arch:
+            log.warning(
+                f"Skipping tests for {self.config.arch}, as the build machine only "
+                f"supports {native_arch}"
+            )
+            return
+
+        if self.build_options.before_test:
+            shell_prepared(self.build_options.before_test, self.build_options, self.env)
+
+        # Install the wheel and test-requires.
+        site_packages_dir = self.tmp_dir / "site-packages"
+        site_packages_dir.mkdir()
+        self.pip_install(
+            "--only-binary=:all:",
+            "--platform",
+            f"android_{self.sysconfigdata['ANDROID_API_LEVEL']}_{self.config.arch}",
+            "--extra-index-url",
+            "https://chaquo.com/pypi-13.1/",
```

Comment:
Why is this needed? Won't it potentially introduce a dependency on "non-official" binary builds?

Reply:
OK, I've removed the `--extra-index-url` from the default command line, and added a note to the documentation explaining how to put it back.